app/components/admin/design_header_component.rb

```diff
@@ -26,38 +28,15 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-require "spec_helper"
-
-RSpec.describe "custom_styles/show" do
-  let(:user) { build(:admin) }
-
-  before do
-    login_as user
-  end
-
-  context "no custom logo yet" do
-    before do
-      assign(:custom_style, CustomStyle.new)
-      assign(:current_theme, "")
-      allow(view).to receive(:options_for_select).and_return("")
-      render
-    end
-
-    it "shows an upload button" do
-      expect(rendered).to include "Upload"
-    end
-  end
-
-  context "with existing custom logo" do
-    before do
-      assign(:custom_style, build(:custom_style_with_logo))
-      assign(:current_theme, "")
-      allow(view).to receive(:options_for_select).and_return("")
-      render
-    end
+module Admin
+  class DesignHeaderComponent < ApplicationComponent
+    include OpPrimer::ComponentHelpers
+    include ApplicationHelper
+    include TabsHelper
 
-    it "shows a replace button" do
-      expect(rendered).to include "Replace"
+    def initialize(tabs: nil)
```

Comment:
Whenever you expect a nullable value, consider using the "blank" alternative to `nil` first.

Here, we expect a list of tabs, but there could be no tabs. Instead of expressing this as `nil`, you could also express this as an empty list. This would help the reader understand what type to expect here. Like so:

```suggestion
    def initialize(tabs: [])
```

Luckily, Rails `present?` method will return `false` for empty Arrays (as well as empty Strings and empty Hashes), so this code change will work the same as `nil` for your purposes.